server/ladders.ts

```diff
@@ -163,7 +163,11 @@ class Ladder extends LadderStore {
 			connection.popup(`You are already challenging someone. Cancel that challenge before challenging someone else.`);
 			return false;
 		}
-		if (targetUser.settings.blockChallenges && !user.can('bypassblocks', targetUser)) {
+		if (
+			user.settings.blockChallenges &&
+			(user.settings.blockChallenges === true || !Users.globalAuth.atLeast(targetUser, user.settings.blockChallenges)) &&
+			!targetUser.can('bypassblocks')
+		) {
```

Comment:
```suggestion
		if (
			targetUser.settings.blockChallenges &&
			(targetUser.settings.blockChallenges === true || !Users.globalAuth.atLeast(user, targetUser.settings.blockChallenges)) &&
			!user.can('bypassblocks', targetUser)
		) {
```
This is backwards; we want to check if the target user is blocking challenges.